platform/platform-configuration-listener/src/test/java/org/codice/ddf/configuration/store/felix/FelixPersistenceStrategyTest.java

```diff
@@ -97,12 +102,82 @@ public void testReadMultiLineFile() throws IOException {
         assertThat(properties.get("key2"), equalTo("value2"));
 
         InOrder inOrder = inOrder(propertyConverter);
-        inOrder.verify(propertyConverter).accept(line1);
-        inOrder.verify(propertyConverter).accept(line2);
+        inOrder.verify(propertyConverter)
+                .accept(line1);
+        inOrder.verify(propertyConverter)
+                .accept(line2);
+    }
+
+    @Test(expected = ConfigurationFileException.class)
+    public void testReadWhenConfigurationHandlerFindsBadValueType() throws Exception {
+        when(propertyConverter.getPropertyNames()).thenReturn(newHashSet("key1", "key2"));
+
+        String line1 = "key1=\"value1\"";
+        String line2 = "key2=Z\"value2\"";
+        String fileContent = line1 + "\r\n" + line2 + "\r\n";
+
+        FelixPersistenceStrategy felixPersistenceStrategy = new FelixPersistenceStrategyUnderTest(
+                fileContent);
+        InputStream inputStream = new ByteArrayInputStream(fileContent.getBytes());
+
+        felixPersistenceStrategy.read(inputStream);
+    }
+
+    /**
+     * Note that Felix' {@link org.apache.felix.cm.file.ConfigurationHandler#read(InputStream)}
+     * does not fail when a boolean's value is wrong, it just sets it to {@code false}.
+     */
+    @Test
+    public void testReadWhenConfigurationHandlerFindsBadBoolean() throws Exception {
+        when(propertyConverter.getPropertyNames()).thenReturn(newHashSet("key1", "key2"));
+
+        String line1 = "key1=\"value1\"";
+        String line2 = "key2=B\"invalid\"";
+        String fileContent = line1 + "\r\n" + line2 + "\r\n";
+
+        FelixPersistenceStrategy felixPersistenceStrategy = new FelixPersistenceStrategyUnderTest(
+                fileContent);
+        InputStream inputStream = new ByteArrayInputStream(fileContent.getBytes());
+
+        Dictionary<String, Object> properties = felixPersistenceStrategy.read(inputStream);
+
+        assertThat(properties, is(notNullValue()));
+        assertThat(properties.get("key1"), equalTo("value1"));
+        assertThat(properties.get("key2"), equalTo(false));
+    }
+
+    @Test(expected = ConfigurationFileException.class)
+    public void testReadWhenConfigurationHandlerFindsBadInteger() throws Exception {
+        when(propertyConverter.getPropertyNames()).thenReturn(newHashSet("key1", "key2"));
+
+        String line1 = "key1=\"value1\"";
+        String line2 = "key2=I\"ABCD\"";
+        String fileContent = line1 + "\r\n" + line2 + "\r\n";
+
+        FelixPersistenceStrategy felixPersistenceStrategy = new FelixPersistenceStrategyUnderTest(
+                fileContent);
+        InputStream inputStream = new ByteArrayInputStream(fileContent.getBytes());
+
+        felixPersistenceStrategy.read(inputStream);
+    }
+
+    @Test(expected = ConfigurationFileException.class)
+    public void testReadWhenConfigurationHandlerFindsBadFloat() throws Exception {
```

Comment:
It would also be nice to test that values with higher cardinalities throw an exception if one of their values are invalid. ie value1=I["1", "two", "3"]